app/services/GandiService.py

```diff
@@ -0,0 +1,51 @@
+import datetime
+
+import requests
+
+
+class GandiService:
+    def __init__(self, config) -> None:
+        self.config = config
+        self.headers = {'Authorization': 'ApiKey ' +
+                        config['keys']['gandi_api_key']}
+        self.params = {'per_page': 1000}
+        self.url = config['urls']['gandi_base_url'] + \
+            config['urls']['gandi_cert_url_extension']
+
+    def get_expiring_certificates(self, email_list):
+        expired_certificates = {}
+        try:
+            certificate_list = requests.get(
+                url=self.url,
+                params=self.params,
+                headers=self.headers)
+            certificate_list.raise_for_status()
+        except requests.exceptions.HTTPError as authentication_error:
+            raise requests.exceptions.HTTPError(
+                f"You may need to export your Gandi API key:\n {authentication_error}")
+        except TypeError as api_key_error:
+            raise TypeError(
+                f"Gandi API key does not exist or is in the wrong format:\n {api_key_error}")
+
```

Comment:
Potential here to have one method for retrieving the entire list, then another for filtering that list down to the expired items - let me know thoughts

Reply:
That's a great idea! Functions should do one thing, though _one thing_ is quite a difficult thing to define 😂 

So any opportunity to simplify the function and separate concerns is good! 😁